src/ast/ctx.rs

```diff
@@ -1166,12 +1178,18 @@ impl<'a, 'ctx> Ctx<'a> {
         Some(s)
     }
 
+    /// # try_set_closure_alloca_bb
+    ///
+    /// it tries to add the closure allocated block into the current context
+    /// if the respective closure captured exists.
+    /// Otherwise, it tries to add this block to its parent context
     pub fn try_set_closure_alloca_bb(&self, bb: BlockHandle) {
         if let Some(c) = &self.closure_data {
             c.borrow_mut().alloca_bb = Some(bb);
         } else if let Some(father) = &self.parent {
             father.try_set_closure_alloca_bb(bb);
         }
+        panic!("{}", "cannot find a context to add the closure")
```

Comment:
wtf are u doing dude?

Reply:
if we cannot find the captured data in all parent context, finally we should panic. that's my idea